### PR TITLE
Upgrade to @variate/engine@2.0.3

### PR DESCRIPTION
**Pull request by ghost**
- Upgrade engine
- Allow user to activate and track from provider render prop return

### CODE REVIEW

File: src/@types/alltypes.d.ts [hunk 1]
```diff
@@ -2,6 +2,7 @@ declare module '@variate/engine';
 
 type ProviderStateType = {
   variate: any
+  audience: object
```
Comment (foxted) [inline src/@types/alltypes.d.ts]: Careful, we renamed audiences to segments recently

File: src/@types/alltypes.d.ts [hunk 2]
```diff
@@ -19,24 +20,24 @@ type VariateComponentProps = {
...
+  experiment: number
   variation: number
 }
 
-interface ComponentReturnInterface {
-  componentName: string,
-  experiments: Array<ExperimentType>,
+type ComponentReturnType = {
+  componentName: string
+  experiments: Array<ExperimentType>
```
Comment (foxted) [inline src/@types/alltypes.d.ts:37]: Is this `ExperimentType` different than the `Experiment` class defined in the engine?

**Review by foxted — CHANGES_REQUESTED** [review]

File: src/@types/alltypes.d.ts (hunk 1 above)
Comment (ghost) [reply src/@types/alltypes.d.ts]: What, when did that happen?

I should change this name.

File: src/@types/alltypes.d.ts (hunk 2 above)
Comment (ghost) [reply src/@types/alltypes.d.ts:37]: Actually, this is a little different.

File: src/@types/alltypes.d.ts (hunk 1 above)
Comment (climbingcode) [reply src/@types/alltypes.d.ts]: I'll change this naming

**Review by foxted — APPROVED** [review]

**Review by foxted — APPROVED** [review]

**Pull request merged**